pkg/yang/types_builtin.go

```diff
@@ -246,9 +246,13 @@ func (e *EnumType) ValueMap() map[int64]string {
 // refer to either a builtin type or type specified with typedef.  Not
 // all fields in YangType are used for all types.
 type YangType struct {
-	Name             string
-	Kind             TypeKind    // Ynone if not a base type
-	Base             *Type       `json:"-"` // dervied from
+	Name string
+	Kind TypeKind // Ynone if not a base type
+	// Base represents the base Type if this is a derived type
+	Base *Type `json:"-"` // dervied from
+	// IdentityBase represents the "base" statement of an Identityref type.
+	// It is distinct from the Base specified above.
+	IdentityBase     *Value
```

Comment:
Try to keep these as line comments, rather than block comments:

Base *Type `json:"-"` // base type for derived types
IdentityBase *Value  // identity base statement for identityrefs